src/client.ts

```
@@ -0,0 +1,196 @@
+  }
+
+  public async verifyAuthentication(): Promise<void> {
+    const endpoint = `${this.serverUrl}/web/api/v2.0/system/info`;
+    try {
+      await makeRequest<any>(endpoint, this.header);
+    } catch (err) {
+      throw new IntegrationProviderAuthenticationError({
+        cause: err,
+        endpoint,
+        status: err.statusCode,
+        statusText: err.message,
+      });
+    }
+  }
+
+  /**
+   * Iterates each agent resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationships
+   */
+  public async iterateAgents(
+    iteratee: ResourceIteratee<SentinelOneAgent>,
+  ): Promise<void> {
+    return this.iterateData(`${this.serverUrl}/web/api/v2.0/agents`, iteratee);
+  }
+
+  /**
+   * Iterates each group resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationships
+   */
+  public async iterateGroups(
+    iteratee: ResourceIteratee<SentinelOneGroup>,
+  ): Promise<void> {
+    return this.iterateData(`${this.serverUrl}/web/api/v2.0/groups`, iteratee);
+  }
+
+  private async iterateData<T>(
+    resourceUrl: string,
+    iteratee: ResourceIteratee<T>,
+  ): Promise<void> {
+    let pagination: SentinelOneApiPagination | undefined;
+    do {
+      const url = paginatedUrl(resourceUrl, pagination);
+      const response = await makeRequest<T>(`${url}`, this.header);
+      pagination = response.pagination;
+
+      if (response.data) {
+        for (const resource of response.data) {
+          await iteratee(resource);
+        }
+      }
+    } while (pagination && pagination.nextCursor);
+  }
+}
+
+export function createAPIClient(
+  config: IntegrationConfig,
+): SentinelOneAPIClient {
+  return new SentinelOneAPIClient(config);
+}
+
+async function makeRequest<T>(
+  url: string | Request,
+  init?: RequestInit,
+): Promise<SentinelOneApiResponse<T>> {
+  const response = await fetch(url, init);
+  if (response.status === 404) {
```

I'm a little concerned that this will silently suppress 404s. Maybe we should at least add a log message here? Are 404s _only_ returned when there is no data to return?

---

I'm unsure, so I kept this as it was. I have seen a few APIs that work that way (include AWS in some services), so I would assume that to be the case here as well.